Lib/pickle.py

```diff
@@ -203,19 +204,45 @@ def commit_frame(self, force=False):
             if f.tell() >= self._FRAME_SIZE_TARGET or force:
                 with f.getbuffer() as data:
                     n = len(data)
-                    write = self.file_write
-                    write(FRAME)
-                    write(pack("<Q", n))
-                    write(data)
+                    po = self.delayed_proto_opcode
+                    if po is None:
+                        self.file_write(FRAME + pack("<Q", n) + data)
```

Comment:
I don't think you should concatenate `data` to the rest. It's not a huge string but it's still 64 kB or more.